### PR TITLE
Buffer tests

### PR DESCRIPTION
**Pull request by qbzzt**
For every opcode with a buffer, check:

1. That it is successful with buffer length 10
2. That it is successful with buffer length 0
3. That it fails with buffer length -10

Try this with different offsets, etc.

### CODE REVIEW

**Comment by holgerd77** [comment]
Was just commenting on the topic "VM tests vs State Tests" in a discussion with @winsvega here https://github.com/ethereumjs/ethereumjs-vm/issues/954#issuecomment-732827621, just as a note here since this fits: I think these kind of tests are generally a good example for test cases which would also fit well being implemented as VM tests since this would allow for more targeted debugging on the VM side.

Generally it's great though that these kind of tests are added (at all) - thanks @qbzzt - these are exactly the implementation nits where client developers (including us) often stumble upon.

**Comment by holgerd77** [comment]
As a follow-up: so - the more I think about that - I have to say that I do like the VM tests as well. 😄

**Comment by qbzzt** [comment]
> Generally it's great though that these kind of tests are added (at all) - thanks @qbzzt - these are exactly the 
> implementation nits where client developers (including us) often stumble upon.

Thank you. This is very good to hear, considering you're our customers. The whole point is to help you develop clients.

File: src/GeneralStateTestsFiller/stMemoryTest/bufferFiller.yml [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: '0x'
+      nonce: '0'
+      storage: {}
+      
+  transaction:
+    data:
+    # SHA3       
+    - :label ok   :abi f(uint,uint) 0x20 0   # Valid buffer length
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stMemoryTest/bufferFiller.yml]: what is the meaning of 0 1 2 10 11 12 12 ?

Comment (qbzzt) [reply src/GeneralStateTestsFiller/stMemoryTest/bufferFiller.yml]: Sorry, the last one was supposed to be 13. I fixed the comment.

File: src/GeneralStateTestsFiller/stMemoryTest/bufferFiller.yml [hunk 2]
```diff
@@ -220,6 +304,15 @@ buffer:
     - :label ok   :abi f(uint,uint) 0x3c 10   # Valid buffer length
     - :label ok   :abi f(uint,uint) 0x3c 11   # Buffer length zero
     - :label fail :abi f(uint,uint) 0x3c 12   # Negative buffer length
+    - :label fail :abi f(uint,uint) 0x3c 20
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stMemoryTest/bufferFiller.yml:307]: i think this need coments too.  20/21/22 the meaning

**Review by winsvega — APPROVED** [review]

**Pull request merged**